src/upgrade/upgrade_scripts/5.14.0/upgrade_bucket_policy.js

```diff
@@ -0,0 +1,53 @@
+/* Copyright (C) 2023 NooBaa */
+"use strict";
+
+const util = require('util');
+const _ = require('lodash');
+const { OP_NAME_TO_ACTION } = require('../../../endpoint/s3/s3_utils');
+
+async function run({ dbg, system_store, system_server }) {
+
+    try {
+        dbg.log0('Starting bucket policy...');
+        const buckets = [];
+        const all_op_names = _.compact(_.flatMap(OP_NAME_TO_ACTION, action => [action.regular, action.versioned]));
+        for (const bucket of system_store.data.buckets) {
+            //Do not update if there are no bucket policy.
+            if (!bucket.s3_policy) return;
+
+            const new_policy = {};
+            if (bucket.s3_policy.version) new_policy.Version = bucket.s3_policy.version;
+
+            new_policy.Statement = bucket.s3_policy.statement.map(statement => ({
+                Effect: statement.effect === 'allow' ? 'Allow' : 'Deny',
+                Action: statement.action.map(atatement_action =>
+                    all_op_names.find(action => action.toLowerCase() === atatement_action)),
```

Comment:
Instead of creating an array and iterate the whole array for each action, create a helper map that looks like the following: 
{ lowercasefunc: CamelCaseFuncName }